### PR TITLE
Resampling (chunk 1/3)

### PR DESCRIPTION
**Pull request by topepo**
I have the resampling chapter fleshed out but decided to break it in to three parts: 

1. Introduction; discussion of general strategies; validation sets; MC-CV; simulation
2. V-fold CV; bootstrapping; niche resampling methods
3. Resamping FAQ

If you want the full firehose, I can delete this PR and give it to you all at once.

### CODE REVIEW

File: chapters/resampling.qmd [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+br_lr <- 
+  stats_bootstraps %>% 
+  filter(model == "logistic" & .metric == "brier_class" & times == 100)
+
+br_1nn <- 
+  stats_bootstraps %>% 
+  filter(model == "knn" & .metric == "brier_class" & times == 100)
+```
+
+In @sec-external-validation, it was shown that using the same data to estimate and evaluate our model can produce inaccurate estimates of how well it functions. It also described using separate partitions of the data to fit and evaluate the model.
```
Comment (kjell-stattenacity) [inline chapters/resampling.qmd:66]:
```suggestion
In @sec-external-validation, it was shown that using the same data to estimate and evaluate our model can produce inaccurate estimates of how well it functions to predict new samples. It also described using separate partitions of the data to fit and evaluate the model. 
```

File: chapters/resampling.qmd [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+  filter(model == "logistic" & .metric == "brier_class" & times == 100)
+
+br_1nn <- 
+  stats_bootstraps %>% 
+  filter(model == "knn" & .metric == "brier_class" & times == 100)
+```
+
+In @sec-external-validation, it was shown that using the same data to estimate and evaluate our model can produce inaccurate estimates of how well it functions. It also described using separate partitions of the data to fit and evaluate the model. 
+
+There are two general approaches for using external data for model evaluation. The first is a validation set, which we’ve seen before in @sec-three-way-split This is a good idea if you have a lot of data on hand. The second approach is to **resample** the training set. Resampling is an iterative approach that reuses the training data multiple times using sound statistical methodology. We’ll discuss both validation sets and resampling in this chapter.
```
Comment (kjell-stattenacity) [inline chapters/resampling.qmd:68]:
```suggestion
There are two general approaches for using external data for model evaluation. The first is a validation set, which we’ve seen before in @sec-three-way-split. This is a good idea if you have a lot of data on hand. The second approach is to **resample** the training set. Resampling is an iterative approach that reuses the training data multiple times based on sound statistical methodology. We’ll discuss both validation sets and resampling in this chapter. 
```

File: chapters/resampling.qmd [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+To understand different methods, let’s examine some theoretical properties of resampling estimates that matter to applied data analysis. 
+
+Let’s start with analogy; consider the usual sample mean estimator used in basic statistics ($\bar{x}$). Based on assumptions about the data, we can derive an equation that optimally estimates the true mean value based on some criterion. Often, the statistical theory focuses on our estimators' theoretical mean and variance. For example, if the data are independent and follow the same Gaussian distribution, the sample mean attempts to estimate the actual population mean (i.e., it is an **unbiased** estimator). We can sometimes derive what the estimator's theoretical variance is, too. These properties, or some combination of them, such as mean squared error^[This discussion is very similar to the variance-bias tradeoff discussed in @sec-variance-bias. The context here differs, but the themes are the same.], help guide us to the best estimator.
```
Comment (kjell-stattenacity) [inline chapters/resampling.qmd:170]:
```suggestion
Let’s start with analogy; consider the usual sample mean estimator used in basic statistics ($\bar{x}$). Based on assumptions about the data, we can derive an equation that optimally estimates the true mean value based on some criterion. Often, the statistical theory focuses on our estimators' theoretical mean and variance. For example, if the data are independent and follow the same Gaussian distribution, the sample mean attempts to estimate the actual population mean (i.e., it is an **unbiased** estimator). Under the same assumptions we can also derive what the estimator's theoretical variance. These properties, or some combination of them, such as mean squared error^[This discussion is very similar to the variance-bias tradeoff discussed in @sec-variance-bias. The context here differs, but the themes are the same.], help guide us to the best estimator. 
```

File: chapters/resampling.qmd [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+
+Let’s start with analogy; consider the usual sample mean estimator used in basic statistics ($\bar{x}$). Based on assumptions about the data, we can derive an equation that optimally estimates the true mean value based on some criterion. Often, the statistical theory focuses on our estimators' theoretical mean and variance. For example, if the data are independent and follow the same Gaussian distribution, the sample mean attempts to estimate the actual population mean (i.e., it is an **unbiased** estimator). We can sometimes derive what the estimator's theoretical variance is, too. These properties, or some combination of them, such as mean squared error^[This discussion is very similar to the variance-bias tradeoff discussed in @sec-variance-bias. The context here differs, but the themes are the same.], help guide us to the best estimator. 
+
+The same is true for resampling methods. We can understand how their bias and variances change under different circumstances and choose an appropriate technique. We'll also focus on bias and variance as the main properties of interest.
```
Comment (kjell-stattenacity) [inline chapters/resampling.qmd:172]:
```suggestion
The same is true for resampling methods. We can understand how their bias and variance change under different circumstances and choose an appropriate technique.  To demonstrate resampling methods, we'll focus on bias and variance as the main properties of interest. 
```

File: chapters/resampling.qmd [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+The same is true for resampling methods. We can understand how their bias and variances change under different circumstances and choose an appropriate technique. We'll also focus on bias and variance as the main properties of interest. 
+
+First is bias: how accurately does a resampling technique estimate the true population value $Q$? We'd like it to be unbiased, but to our knowledge, that is nearly impossible. However, some things we do know.  @fig-resampling-scheme shows that the training set (of size $n_{tr}$) is split into two partitions of size $n_{fit}$ and $n_{pred}$. It turns out that, as $n_{pred}$ becomes larger, the bias increases in a pessimistic direction^[Meaning that performance looks worse than it should. For example, smaller $R^2$ or inflated Brier score.]. In other words, if our training set has 1,000 samples, a resampling method where the assessment set has $n_{pred} = 100$ data points has a smaller bias than one using $n_{pred} = 500$. Another thing that we know is that increasing the number of resamples (_B_) can't significantly reduce the bias.
```
Comment (kjell-stattenacity) [inline chapters/resampling.qmd:174]:
```suggestion
First is bias: how accurately does a resampling technique estimate the true population value $Q$? We'd like it to be unbiased, but to our knowledge, that is nearly impossible. However, some things we do know.  @fig-resampling-scheme shows that the training set (of size $n_{tr}$) is split into _B_ resampling partitions which are then split into analysis and assessment sets of size $n_{fit}$ and $n_{pred}$, respectively. It turns out that, as $n_{pred}$ becomes larger, the bias increases in a pessimistic direction^[Meaning that performance looks worse than it should. For example, smaller $R^2$ or inflated Brier score.]. In other words, if our training set has 1,000 samples, a resampling method where the assessment set has $n_{pred} = 100$ data points has a smaller bias than one using $n_{pred} = 500$. Another thing that we know is that increasing the number of resamples (_B_) can't significantly reduce the bias.  Therefore, we must think carefully about the way the data is partitioned in the resamplint process to reduce potential bias.
```

File: chapters/resampling.qmd [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+  geom_hline(yintercept = 0, lty = 2) +
+  labs(y = "Std. Error", x = "Number of Resamples") +
+  scale_color_brewer(palette = "Set1")
+
+mc_bias + mc_prec +
+  plot_layout(guides = 'collect') &
+  theme(legend.position = "top")
+```
+
+The results of this simulation indicate that, in terms of precision, there is little benefit in including more than 70% of the training set in the assessment set. Also, while more resamples always help, there is incremental benefit in using more than 50 or 60 resamples (for this size training set). Regarding bias, the decrease somewhat slows down near 80% of the training set retained. Holdout proportions of around 75% - 80% might be a reasonable rule of thumb (these are, of course, subjective).
```
Comment (kjell-stattenacity) [inline chapters/resampling.qmd:264]:
```suggestion
The results of this simulation indicate that, in terms of precision, there is little benefit in including more than 70% of the training set in the assessment set. Also, while more resamples always help, there is incremental benefit in using more than 50 or 60 resamples (for this size training set). Regarding bias, the decrease somewhat slows down near 80% of the training set retained. Holdout proportions of around 75% - 80% might be a reasonable rule of thumb (these are, of course, subjective and based on this simulation).
```

**Review by kjell-stattenacity — CHANGES_REQUESTED** [review]
